src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.utils.tsx

```diff
@@ -0,0 +1,79 @@
+import styles from './dataset-info.module.css'
+
+import cn from 'classnames'
+
+import { CardTitle } from '@ui/card-main'
+import { IDsInfoReceipt } from '@service-providers/dataset-level'
+import { Row } from './dataset-info.interfaces'
+
+export const renderRow =
+  <T,>(info: T) =>
+  (it: Row<keyof T>) => {
+    const value = info[it.field]
+
+    if (it.optional && (!value || (it.render && !it.render(value)))) {
+      return null
+    }
+
+    return (
+      <tr key={it.fieldName} className={cn(styles.datasetInfo__table_row)}>
+        <td className={cn(styles.datasetInfo__table_row_title)}>
+          {it.fieldName}
+        </td>
+        <td>{it.render ? it.render(value) : value}</td>
+      </tr>
+    )
+  }
+
+export const renderAncestor = (it: IDsInfoReceipt) => {
+  const isFilter = it.kind === 'filter'
+
+  const title = `${
+    isFilter ? 'Filter applied in' : 'Decision tree code applied in'
```

Comment:
I suppose copy text should be moved to en.ts

Reply:
Add localization everywhere